tests/behat/context/ApiContext.php

```diff
@@ -1635,6 +1650,271 @@ public function iShouldGetTheStoredJsonObject(string $name): void
+    {
+      $values = [];
+      foreach ($returned_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($returned_programs), 0, 'Returned projects have ['.$projects.'] unmatched');
+    }
+    if (count($expected_programs) > 0)
+    {
+      $values = [];
+      foreach ($expected_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($expected_programs), 0, 'Expected projects have ['.$projects.'] unmatched');
+    }
+  }
+
+  /**
+   * @Then /^The response should contain the following project:$/
+   */
+  public function responseShouldContainTheFollowingProject(TableNode $table): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+
+    $returned_programs = json_decode($response->getContent(), true);
+    $expected_programs = $table->getHash();
+
+    Assert::assertEquals(count($returned_programs), count($expected_programs), 'Number of returned programs should be '.count($returned_programs));
+
+    foreach ($expected_programs as $expected_key => $expected_program)
+    {
+      foreach ($returned_programs as $returned_key => $returned_program)
+      {
+        if ($expected_program['Name'] === $returned_program['name'])
+        {
+          unset($expected_programs[$expected_key], $returned_programs[$returned_key]);
+
+          break;
+        }
+      }
+    }
+    unset($expected_program, $returned_program);
+
+    if (count($returned_programs) > 0)
+    {
+      $values = [];
+      foreach ($returned_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($returned_programs), 0, 'Returned projects have ['.$projects.'] unmatched');
+    }
+    if (count($expected_programs) > 0)
+    {
+      $values = [];
+      foreach ($expected_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($expected_programs), 0, 'Expected projects have ['.$projects.'] unmatched');
+    }
+  }
+
+  /**
+   * @Then /^the response should contain projects in the following order:$/
+   */
+  public function responseShouldContainProjectsInTheFollowingOrder(TableNode $table): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+
+    $responseArray = json_decode($response->getContent(), true);
+    $returned_programs = $responseArray['projects'];
+    $expected_programs = $table->getHash();
+
+    Assert::assertEquals(count($returned_programs), count($expected_programs), 'Number of returned programs should be '.count($returned_programs));
+
+    for ($i = 0; $i < count($returned_programs); ++$i)
+    {
+      Assert::assertEquals($returned_programs[$i]['name'], $expected_programs[$i]['Name']);
```

Comment:
we still only compare the name -
we should really also check the values of all other fields.

My suggestion: 

- don't define the name in the feature file - use the unique id
- get the project from the database
- for every structure key check the value with the returned one
- some fields like the urls might need a little extra logic